synapse/handlers/groups_local.py

```diff
@@ -37,9 +37,18 @@ def f(self, group_id, *args, **kwargs):
             )
         else:
             destination = get_domain_from_id(group_id)
-            return getattr(self.transport_client, func_name)(
+            d = getattr(self.transport_client, func_name)(
                 destination, group_id, *args, **kwargs
             )
+
+            def h(failure):
```

Comment:
a comment about what we're doing here wouldn't go amiss.

Reply:
added some longish comments